### PR TITLE
Enhancements for UID, TLM, found/lost, and distance

### PR DESCRIPTION
**Pull request by sandeepmistry**
Includes #9 and the following:

* TLM data is OR’ed with old, if present
* events changed to ``discover``, ``found``, and ``lost``
* use simplified distance formula for now (same as examples in google/eddystone repo)
* only emit ``lost`` when duplicates is enabled

@christopherdro @jacobrosenthal @ukBaz comments and feedback welcome ...

### CODE REVIEW

File: lib/eddystone-beacon-scanner.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
 var events = require('events');
 var util = require('util');
 
-var debug = require('debug')('eddystone-beacon-scanner');
+var debug = require('debug')('eddstone-beacon-scanner');
```
Comment (jacobrosenthal) [inline lib/eddystone-beacon-scanner.js]: Misspelled eddystone

**Comment by christopherdro** [comment]
:+1: Just testing it out and it looks great! Awesome work.

File: README.md [hunk 2]
```diff
@@ -23,20 +21,55 @@ See [examples](examples) folder.
...
+
+##### UID
+ * ```type``` - Eddystone type
+ * ```txPower``` - Measured received power at 0 m in dBm
+ * ```namespace``` - 10-byte ID Namespace
+ * ```instance``` - 6-byte ID Instance
+ * ```rssi``` - RSSI of discovered beacon
+ * ```distance``` - Approximate distance from beacon
+
+##### TLM
```
Comment (jacobrosenthal) [inline README.md]: I think we need some kind of language explaining that if uid interleaving is being used, you wont receive a tlm type, but rather a uid type with tlm props added.

Comment (sandeepmistry) [inline README.md]: @jacobrosenthal agreed, I was thinking about moving all TLM fields to a ```tlm``` property:

Comment (christopherdro) [inline README.md]: @sandeepmistry I agree, that sounds like a good idea.

File: lib/eddystone-beacon-scanner.js (hunk 1 above)
Comment (sandeepmistry) [inline lib/eddystone-beacon-scanner.js]: Thanks, updated

File: README.md (hunk 2 above)
Comment (sandeepmistry) [inline README.md]: Cool, I've split it out and also updated the read me.

@jacobrosenthal any feedback on the updated read me?

**Comment by jacobrosenthal** [comment]
Nice work @sandeepmistry @christopherdro

**Comment by christopherdro** [comment]
Nice! Thanks for taking the time for your great additions @sandeepmistry @jacobrosenthal

**Pull request merged**